src/langcheck/metrics/eval_clients/_swallow.py

```diff
@@ -0,0 +1,230 @@
+            response.outputs[0].text if response else None
+            for response in responses
+        ]
+
+        return response_texts
+
+    def get_float_score(
+        self,
+        metric_name: str,
+        language: str,
+        unstructured_assessment_result: list[str | None],
+        score_map: dict[str, float],
+    ) -> list[float | None]:
+        """The function that transforms the unstructured assessments (i.e. long
+        texts that describe the evaluation results) into scores.
+
+        Args:
+            metric_name : The name of the metric to be used. (e.g. "toxicity")
+            language: The language of the prompts. (e.g. "en")
+            unstructured_assessment_result: The unstructured assessment results
+                for the given assessment prompts.
+            score_map: The mapping from the short assessment results
+                (e.g. "Good") to the scores.
+        Returns:
+            A list of scores for the given prompts. The scores can be None if
+            the evaluation fails.
+        """
+        if language not in ["en", "ja"]:
+            raise ValueError(f"Unsupported language: {language}")
+
+        options = list(score_map.keys())
+        get_score_template = get_template(f"{language}/get_score/plain_text.j2")
+        get_score_prompts = [
+            get_score_template.render(
+                {
+                    "metric": metric_name,
+                    "unstructured_assessment": unstructured_assessment,
+                    "options": options,
+                }
+            )
+            if unstructured_assessment
+            else None
+            for unstructured_assessment in unstructured_assessment_result
+        ]
+
+        messages = [
+            [
+                {
+                    "role": "system",
+                    "content": self._system_prompts[language],
+                },
+                {
+                    "role": "user",
+                    "content": prompt,
+                },
+            ]
+            for prompt in get_score_prompts
+        ]
+        prompts = self._tokenizer.apply_chat_template(
+            messages, tokenize=False, add_generation_prompt=True
+        )
+        if isinstance(prompts, str):
+            prompts = [prompts]
+        else:
+            prompts = [str(p) for p in prompts]
+        responses = self._model.generate(prompts, self._sampling_params)
+
+        raw_response_texts = [
+            response.outputs[0].text if response else None
+            for response in responses
+        ]
+
+        def _turn_to_score(response: str | None) -> float | None:
+            if response is None:
+                return None
+            option_found = [option for option in options if option in response]
+            # if response contains multiple options as substrings, return None
+            if len(option_found) != 1:
+                return None
+            return score_map[option_found[0]]
```

Comment:
The "turn to score" logic in Prometheus (i.e. find the option that appears last) looks more robust to me - let's use that here as well

Reply:
(Ah nevermind, just noticed that we're using function calling here so it makes sense that only one should be outputted!)